SpatialGDK/Source/SpatialGDK/Private/Utils/InterestFactory.cpp

```diff
@@ -16,93 +17,54 @@
+// It is built once per net driver initialisation.
+static QueryConstraint ClientCheckoutRadiusConstraint;
 
-	const AActor* DefaultActor = Cast<AActor>(AActor::StaticClass()->GetDefaultObject());
-	const float DefaultDistanceSquared = DefaultActor->NetCullDistanceSquared;
-	const float MaxDistanceSquared = GetDefault<USpatialGDKSettings>()->MaxNetCullDistanceSquared;
+InterestFactory::InterestFactory(AActor* InActor, const FClassInfo& InInfo, USpatialClassInfoManager* InClassInfoManager, USpatialPackageMapClient* InPackageMap)
+	: Actor(InActor)
+	, Info(InInfo)
+	, ClassInfoManager(InClassInfoManager)
+	, PackageMap(InPackageMap)
+{
+}
 
-	// Gather ClientInterestDistance settings, and add any larger than the default radius to a list for processing.
-	TMap<UClass*, float> DiscoveredInterestDistancesSquared;
-	for (TObjectIterator<UClass> It; It; ++It)
-	{
-		if (It->HasAnySpatialClassFlags(SPATIALCLASS_ServerOnly))
-		{
-			continue;
-		}
-		if (!It->HasAnySpatialClassFlags(SPATIALCLASS_SpatialType))
-		{
-			continue;
-		}
-		if (It->HasAnyClassFlags(CLASS_NewerVersionExists))
-		{
-			// This skips classes generated for hot reload etc (i.e. REINST_, SKEL_, TRASHCLASS_)
-			continue;
-		}
-		if (!It->IsChildOf<AActor>())
-		{
-			continue;
-		}
+void InterestFactory::CreateClientCheckoutRadiusConstraint(USpatialClassInfoManager* ClassInfoManager)
+{
+	// Checkout Radius constraints are defined by the NetCullDistanceSquared property on actors.
+	//   - Checkout radius is a RelativeCylinder constraint on the player controller.
+	//   - NetCullDistanceSquared on AActor is used to define the default checkout radius with no other constraints.
+	//   - NetCullDistanceSquared on other actor types is used to define additional constraints if needed.
+	//   - If a subtype defines a radius smaller than a parent type, then its requirements are already captured.
+	//   - If a subtype defines a radius larger than all parent types, then it needs an additional constraint.
+	//   - Other than the default from AActor, all radius constraints also include Component constraints to
+	//     capture specific types, including all derived types of that actor.
 
-		const AActor* IteratedDefaultActor = Cast<AActor>(It->GetDefaultObject());
-		if (IteratedDefaultActor->NetCullDistanceSquared > DefaultDistanceSquared)
-		{
-			float ActorNetCullDistanceSquared = IteratedDefaultActor->NetCullDistanceSquared;
+	QueryConstraint CheckoutRadiusConstraint;
 
-			if (MaxDistanceSquared != 0.f && IteratedDefaultActor->NetCullDistanceSquared > MaxDistanceSquared)
-			{
-				UE_LOG(LogInterestFactory, Warning, TEXT("NetCullDistanceSquared for %s too large, clamping from %f to %f"),
-					*It->GetName(), ActorNetCullDistanceSquared, MaxDistanceSquared);
+	CheckoutRadiusConstraint.OrConstraint.Add(CheckoutRadiusConstraintUtils::GetDefaultCheckoutRadiusConstraint());
```

Comment:
Shouldn't this constraint be AND with something that indicates it's an Unreal GDK entity? Like the position?

Reply:
I'm not sure I understand- all entities would have the position component right?

Reply:
Oh, I mean a gdk only component. Perhaps UnrealMetadata? Just seems strange that we're grabbing all entities here. I do see that's what it used to be however. Would there be a performance hit if we were to add the additional component constraint?

Reply:
that makes sense- I wasn't sure when writing this if there was never a case when a client would want to check out a non-unreal entity. The performance hit for a change like this would be minimal, as it _should_ be a quick lookup and only ever executed if the spatial constraint held true.

Reply:
Happy to add this if you're at least 80% sure it's fine.

Reply:
I agree with the first point, but the second one makes this seem risky- I would rather raise a ticket and make this change in a separate PR in case it breaks things, as ideally this PR would only do the one thing

Reply:
Actually, does the level constraint not do this? Do level components get added to non unreal entities?

Reply:
@jacquese Ahh yes, you're right. Good to leave as is then 👍